packages/sdk/test/evm/new-publish-deploy.test.ts

```diff
@@ -0,0 +1,270 @@
+      );
+      await mockUploadWithCID("bytecode1", compilerMetadata.bytecode);
+      await mockUploadWithCID(
+        "cid1",
+        JSON.stringify({
+          ...extendedMetadata,
+          metadataUri: compilerMetadata.metadataUri,
+          bytecodeUri: "bytecode1",
+          networksForDeployment: {
+            allNetworks: true,
+            networksEnabled: [],
+          },
+          version: "1.0.2",
+        }),
+      );
+
+      // deploy custom-factory `SplitMain`
+      const customFactoryAddress = await sdk.deployer.deployContractFromUri(
+        "cid1",
+        [],
+      );
+
+      // Re-upload contract metadata with mock storage
+      // set custom-factory address on hardhat chain
+      const contractMetadata = await fetchAndCacheDeployMetadata(
+        publishUriWithCustomFactoryDeploy,
+        realSDK.storage,
+      );
+      await mockUploadWithCID(
+        contractMetadata.compilerMetadata.metadataUri.replace("ipfs://", ""),
+        JSON.stringify(contractMetadata.compilerMetadata.metadata),
+      );
+      await mockUploadWithCID(
+        "bytecode2",
+        contractMetadata.compilerMetadata.bytecode,
+      );
+      await mockUploadWithCID(
+        "cid2",
+        JSON.stringify({
+          ...contractMetadata.extendedMetadata,
+          metadataUri: contractMetadata.compilerMetadata.metadataUri,
+          bytecodeUri: "bytecode2",
+          factoryDeploymentData: {
+            ...contractMetadata.extendedMetadata?.factoryDeploymentData,
+            implementationAddresses: {},
+            customFactoryInput: {
+              factoryFunction: "createSplit",
+              params: [],
+              customFactoryAddresses: {
+                "31337": customFactoryAddress,
+              },
+            },
+          },
+          networksForDeployment: {
+            allNetworks: true,
+            networksEnabled: [],
+          },
+          version: "1.0.3",
+        }),
+      );
+
+      // deploy contract via custom-factory
+      const contractAddress = await sdk.deployer.deployContractFromUri("cid2", [
+        [await bobWallet.getAddress(), await adminWallet.getAddress()],
+        [500000, 500000],
+        0,
+        await adminWallet.getAddress(),
+      ]);
+
+      const splitContract = await sdk.getContract(contractAddress);
+      const deployingAddress = await splitContract.call("splitMain");
+
+      expect(deployingAddress).to.equal(customFactoryAddress);
+    });
+  });
+
+  describe("Deploy re-published contract", async () => {
+    let publishUriWithAutoFactoryDeploy: string;
+
+    before("", async () => {
+      // Multiwrap
+      publishUriWithAutoFactoryDeploy =
+        "ipfs://QmQBgncSxFwLKusgAzXVpaWTuJTfV5yL6vs3jYXzz89EaY/";
+    });
+
+    it("should deploy with auto factory", async () => {
+      const mockPublisher = process.env.contractPublisherAddress;
+      process.env.contractPublisherAddress =
+        "0x664244560eBa21Bf82d7150C791bE1AbcD5B4cd7";
```

Comment:
why did you need this?

Reply:
when fetching a published contract for any-evm, it queries the polygon publisher